WebtoHunspell/WebtoHunspell.hs

```diff
@@ -0,0 +1,1209 @@
+{-=Spellit=-}
+{-=WebtoHunspell: A .aff File Creator.=-}
+{-=Author: Matthew Mosior=-}
+{-=Version: Pre-Alpha=-}
+{-=Synopsis:  This Haskell Script will create a affix file=-}
+{-=from the JSON input files.=-}
+
+{-Syntax Extensions.-}
+
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE QuasiQuotes       #-}
+
+{--------------------}
+
+{-Imports-}
+
+import Control.Monad (mzero)
+import Data.Aeson as A
+import Data.Aeson.Types as AT
+import Data.Aeson.Internal as ATI
+import Data.ByteString.Lazy.Char8 as LBSC
+import Data.Char as Ch
+import Data.Function as Fn
+import Data.HashMap.Lazy as HL
+import Data.List as L
+import Data.List.Split as S
+import qualified Data.List.Extra as E
+import Data.Maybe as M
+import Data.Ord as Or
+import Data.Text as T
+import System.Directory as D
+import System.Environment as En
+import System.IO as IO
+import System.IO.Temp as Temp
+import Text.PrettyPrint.Boxes as PB
+
+{---------}
+
+{-Custom Data Declarations (JSON).-}
+
+--Paradigm
+data Paradigm = Paradigm {
+    paradigm :: String
+  , slots :: [String]
+  , words :: A.Value
+  } deriving (Read,Show)
+
+instance FromJSON Paradigm
+    where
+        parseJSON (Object v) = Paradigm <$> (v .: "paradigm_name")
+                                        <*> (v .: "slots")
+                                        <*> (v .: "words")
+
+        parseJSON _          = mzero
+--
+
+--Inputs
+data Inputs = Inputs {
+    slotandword :: [(String,String)]
+    } deriving (Read,Show)
+--
+--
+
+{----------------------------------}
+
+{-Affix file-}
+{-Custom Data Declarations (after JSON).-}
+
+data AffixFileEncoding = 
+    AffixFileEncoding String
+    deriving (Eq,Ord,Show)
+
+data PreRuleHeader =
+    PrePfxHead String String Char
+  | PreSfxHead String String Char
+    deriving (Eq,Ord,Show) 
+
+data PreRule =
+    PrePfx String String String String
+  | PreSfx String String String String
+    deriving (Eq,Ord,Show)
+
+data RuleHeader =
+    PfxHead String String Char Int
+  | SfxHead String String Char Int
+    deriving (Eq,Ord,Show)
+
+data Rule = 
+    Pfx String String String String String 
+  | Sfx String String String String String
+    deriving (Eq,Ord,Show)
+
+{----------------------------------------}
+{------------}
+
+{-Dictionary file-}
+{-Custom Data Declarations (after JSON).-}
+
+data DictFileHeader = 
+    DictFileHeader String
+    deriving (Eq,Ord,Show)
+
+data DictFileBody = 
+    DictFileBody String
+    deriving (Eq,Ord,Show)
+
+{----------------------------------------}
+{-----------------}
+
+{-General Utility Functions.-}
+
+--linefeed -> To feed file into nested list line by line, 
+--delimited by newline characters.
+linefeed :: String -> [String]
+linefeed xs = L.lines xs
+
+--singleunnest
+singleunnest :: [a] -> a
```

Comment:
Sorry but what dos the "unnest" and "mtriple" stand for?

Reply:
These are names of functions.  I tried to convey the functionality contained within the function in its name.